recipes-azure-iot/iot-hub-device-update/iot-hub-device-update/swupdate_handler_v2_grub.sh

```diff
@@ -0,0 +1,757 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Ensure that getopt starts from first option if ". <script.sh>" was used.
+OPTIND=1
+
+ret_val=0
+
+# Ensure we dont end the user's terminal session if invoked from source (".").
+if [[ $0 != "${BASH_SOURCE[0]}" ]]; then
+    ret='return'
+else
+    ret='exit'
+fi
+
+# Output formatting.
+
+# Log level: 0=debug, 1=info, 2=warning, 3=error, 4=none
+log_level=1
+
+warn() { echo -e "\033[1;33mWarning:\033[0m $*" >&2; }
+
+error() { echo -e "\033[1;31mError:\033[0m $*" >&2; }
+
+header() { echo -e "\e[4m\e[1m\e[1;32m$*\e[0m"; }
+
+bullet() { echo -e "\e[1;34m*\e[0m $*"; }
+
+# Log debug prefix - blue
+log_debug_pref="\033[1;30m[D]\033[0m"
+
+# Log info prefix - blue
+log_info_pref="\033[1;34m[I]\033[0m"
+
+# Log warning prefix - yellow
+log_warn_pref="\033[1;33m[W]\033[0m"
+
+# Log error prefix - red
+log_error_pref="\033[1;31m[E]\033[0m"
+
+#
+# Files and Folders information
+#
+workfolder=
+output_file=/var/log/aduc-logs/swupdatehandlerv2.output
+log_file=/var/log/aduc-logs/swupdatehandlerv2.log
+swupdate_log_file=/var/log/aduc-logs/swupdate.log
+result_file=/var/log/aduc-logs/swupdatehandlerv2.result
```

Comment:
can we add timestamp suffixes ... so i don't overwrite logs when doing multiple runs of install, e.g. without apply?

Reply:
imho, the log files won't be overwritten. The script will always append the log info e.g. "echo "xxx" **>>** "

Reply:
does upstream use multiple log files, or did we introduce that or added to that? `log_file` vs `swupdate_log_file` (vs. `output_file`)?